cognitedata/cdf-fabric-replicator
# Added event replicator and unit tests

## What is this PR for?
Added code for event replicator that will replicate event data from CDF and store in a Lakehouse table, with the table path and name configurable.

Also added unit tests for the event replicator.

## What does the code do?
- The code for the event replicator is in "event.py".
- Adjusted config to incorporate configurations for event replicator including batch size and abfss table path
- Added unit tests for the replicator in "tests/unit/test_event.py"

## How do I verify the changes made?
- Copy the ".env.example" to ".env" and populate proper values for all environment variables. The environment variable that's specifically for event replicator is `EVENT_TABLE_NAME`
- The default batch size in "example_config.yaml" is set to 5. Change the batch size if needed.
- [optional] In "__main__.py", comment out thread creation for other replicators to focus resources on the event replicator
- Run `poetry run cdf_fabric_replicator example_config.yaml` to run the event replicator

<img width="1202" alt="image" src="https://github.com/cognitedata/cdf-fabric-replicator/assets/123187956/a93b1e99-bcbe-41be-a057-b3ba97dfcaef">

<img width="1880" alt="image" src="https://github.com/cognitedata/cdf-fabric-replicator/assets/123187956/bc60a966-60fd-473d-a283-8e1d7f79757d">


- Run `poetry run pytest tests/unit/test_event.py` to verify that all unit tests pass

<img width="1157" alt="image" src="https://github.com/cognitedata/cdf-fabric-replicator/assets/123187956/4914b757-e58d-41af-8422-0180323a648a">
